R/step_adjust_latency.R

```diff
@@ -0,0 +1,274 @@
+           method = c(
+             "extend_ahead",
+             "locf",
+             "extend_lags"
+           ),
+           fixed_latency = NULL,
+           fixed_asof = NULL,
+           default = NA,
+           skip = FALSE,
+           columns = NULL,
+           id = recipes::rand_id("epi_lag")) {
+    arg_is_chr_scalar(id, method)
+    if (!is_epi_recipe(recipe)) {
+      cli::cli_abort("This recipe step can only operate on an `epi_recipe`.")
+    }
+    if (!is.null(columns)) {
+      cli::cli_abort(c("The `columns` argument must be `NULL`.",
+        i = "Use `tidyselect` methods to choose columns to lag."
+      ))
+    }
+
+    method <- rlang::arg_match(method)
+    terms_used <- recipes_eval_select(enquos(...), recipe$template, recipe$term_info)
+    if (length(terms_used) == 0) {
+      terms_used <- recipe$term_info %>%
+        filter(role == "raw") %>%
+        pull(variable)
+    }
+    if (method == "extend_ahead") {
+      rel_step_type <- "step_epi_ahead"
+      shift_name <- "ahead"
+    } else if (method == "extend_lags") {
+      rel_step_type <- "step_epi_lag"
+      shift_name <- "lag"
+    }
+    relevant_shifts <- construct_shift_tibble(terms_used, recipe, rel_step_type, shift_name)
+
+    if (!any(map_lgl(
+      recipe$steps,
+      function(recipe_step) inherits(recipe_step, rel_step_type)
+    ))) {
+      cli:cli_abort("there is no `{rel_step_type}` defined before this. for the method `extend_{shift_name}` of `step_adjust_latency`, at least one {shift_name} must be previously defined.")
+    }
+
+    recipes::add_step(
+      recipe,
+      step_adjust_latency_new(
+        terms = dplyr::enquos(...),
+        role = role,
+        method = method,
+        trained = trained,
+        as_of = fixed_asof,
+        latency = fixed_latency,
+        shift_cols = relevant_shifts,
+        default = default,
+        keys = epi_keys(recipe),
+        skip = skip,
+        id = id
+      )
+    )
+  }
+
+step_adjust_latency_new <-
+  function(terms, role, trained, as_of, latency, shift_cols, time_type, default,
+           keys, method, skip, id) {
+    step(
+      subclass = "adjust_latency",
+      terms = terms,
+      role = role,
+      method = method,
+      trained = trained,
+      as_of = as_of,
+      latency = latency,
+      shift_cols = shift_cols,
+      default = default,
+      keys = keys,
+      skip = skip,
+      id = id
+    )
+  }
+
+# lags introduces max(lags) NA's after the max_time_value.
+# TODO all of the shifting happens before NA removal, which saves all the data I might possibly want; I should probably add a bit that makes sure this operation is happening before NA removal so data doesn't get dropped
+#' @export
+prep.step_adjust_latency <- function(x, training, info = NULL, ...) {
+  if ((x$method == "extend_ahead") && (!("outcome" %in% info$role))) {
+    cli::cli_abort('If `method` is `"extend_ahead"`, then a step ",
+          "must have already added an outcome .')
```

Comment:
```suggestion
cli::cli_abort(c(
  "If `method` is {.val extend_ahead}, then a step ",
  "must have already added an outcome."
))
```

Reply:
what does `{.val` do? in this case I was trying to make a R code string in the docs, like this: `"extend_ahead"`